### PR TITLE
Make jax.numpy.squeeze as strict as numpy.squeeze about axis shape.

### PR DESCRIPTION
**Pull request by GeorgOstrovski**
Raise error if an axis explicitly selected to be squeezed has shape != 1.

### CODE REVIEW

**Comment by GeorgOstrovski** [comment]
Related issue: #2284

**Comment by GeorgOstrovski** [comment]
Apologies, didn't test this properly - will fix and re-open then.
